### PR TITLE
feat: new cli commands `accounts:notes` and `accounts:transactions`

### PR DESCRIPTION
**Pull request by wd021**
Proposing to add an `accounts:transactions` command to display transaction notes for an account. I've been debugging some account balance issues lately and found this to be quite helpful. Here's what it looks like

Screen Shot 2022-02-28 at 2 04 16 PM

If we add this, I'll open a PR to update docs on the website repo.

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by dguenther** [comment]
Going to assign this to @NullSoldier because I think he wanted to give it a review next week before it merges. This looks useful!

**Comment by QustomQure** [comment]
I think that this command must be able to work with both accountName and account address for better usability

**Comment by wd021** [comment]
for consistency, I'm using the same format as `accounts:balance`, which is accountName. We can update to take both, although it is one quick step to get names from `accounts:list`

**Comment by Allmaaz** [comment]
Very good project

**Comment by wd021** [comment]
@dguenther @NullSoldier still interested in merging this? 

If so, i can update the code for latest version.. lmk!

**Comment by dguenther** [comment]
I'd still like to merge this, I think it's definitely useful especially in the latest testnet phase. Nobody on our side has been directly involved in reviewing + merging it, so sorry for the delay! I'd like to get one other person than myself to take a look at it, then it should be good to go, if you don't mind updating the code to fix conflicts.

File: ironfish/src/account/accounts.ts
Comment (rohanjadvani) [inline ironfish/src/account/accounts.ts:637]: Is this `continue` needed?

File: ironfish-cli/src/commands/accounts/transactions.test.ts
Comment (xorinox) [inline ironfish-cli/src/commands/accounts/transactions.test.ts:5]: This import didn't work for me, I had to change the path to  `../../../../ironfish/src/rpc/routes/transactions`

Comment (wd021) [reply ironfish-cli/src/commands/accounts/transactions.test.ts]: yup, this needs updating for the latest version. I just pushed an update.

File: ironfish/src/account/accounts.ts
Comment (wd021) [reply ironfish/src/account/accounts.ts]: nope, removed.

**Comment by mat-if** [comment]
Hey @wd021, I'm starting to dig into this PR, sorry for the delay. I wanted to clarify which direction to look at this from.

What's the intended goal of this CLI command? I think there's a few different things technically happening here so I wanted to make sure we're on the same page on expectations of what questions this command should answer.

For example, are you expecting this to show a complete list of all transactions this account has participated in? A complete list of notes, spent and unspent? A list of transactions containing unspent notes? etc.

Looking at this right now, it feels a little bit awkward in that it is kind of a view into transactions, and also kind of a view into notes, and I think we can improve that together

**Comment by wd021** [comment]
@mat-if no worries. 

The motivation for adding this command was to help debug account balances & faucet errors during phase 1. I wanted an easy way to output the spent notes associated with a given account.

I agree that the wording is probably unclear. Would also be interested in expanding the scope of this into filtering by txs/notes/unspent notes.

**Comment by wd021** [comment]
yea, 1 and 2 were exactly where my mind was at. let's have 2 separate methods, `accounts:transactions` and `accounts:notes`. And also agree that 3 could be useful too!

i don't mind taking a crack at all 3 in this PR, will have some time later in the week 👨‍💻

**Comment by wd021** [comment]
@mat-if PR updated. here's the latest👇 

`accounts:notes` - note level breakdown for given account (optional `-a` flag)

Screenshot 2022-06-06 at 11 08 36

`accounts:transactions` - transaction breakdown for given account or txhash (optional `-a` / `-t` flags)

Screenshot 2022-06-06 at 11 08 56

Screenshot 2022-06-06 at 11 09 33

File: ironfish-cli/src/commands/accounts/transactions.ts
Comment (mat-if) [inline ironfish-cli/src/commands/accounts/transactions.ts:52]: What do you think about making each of these on a new line? I think it makes it a little easier to parse the info.
```
Transaction: fdaf29274587c8f58ac6214f42642fd610a222cea8e5e39ff4e11d58d379c194
Status: completed
Miner Fee: x
Fee ($ORE): 1
Spends: 1

```

File: ironfish/src/account/accounts.ts
Comment (mat-if) [inline ironfish/src/account/accounts.ts:1043]: Same thing here about the `transactionCreator` thing, not able to look up a transaction that a note references if I only received in that transaction

image

**Review by mat-if — CHANGES_REQUESTED** [review]
Thanks for doing this! A few comments, but overall this looks pretty good.

File: ironfish/src/account/accounts.ts
Comment (wd021) [reply ironfish/src/account/accounts.ts]: oh right, good catch.

File: ironfish/src/account/accounts.ts
Comment (wd021) [reply ironfish/src/account/accounts.ts]: makes sense 👌

File: ironfish/src/account/accounts.ts
Comment (wd021) [reply ironfish/src/account/accounts.ts]: makes sense 👌

File: ironfish-cli/src/commands/accounts/transactions.ts
Comment (wd021) [reply ironfish-cli/src/commands/accounts/transactions.ts]: update to this

File: ironfish/src/account/accounts.ts
Comment (wd021) [reply ironfish/src/account/accounts.ts]: right. I updated this method to include notes where one is the receiver of iron in a transaction. but i also think its valuable to be able to detect the transactions created by the account, so there's now an extra `creator` column for the table.

File: ironfish/src/account/accounts.ts
Comment (wd021) [reply ironfish/src/account/accounts.ts]: yup, removed this.

**Comment by wd021** [comment]
@mat-if 
- went through review changes and updated
- added extra `creator` column for account:transactions
- merge latest staging changes to resolve merge conflict

**Review by mat-if — APPROVED** [review]
Thanks!

**Pull request merged**
